File: Admin_p/Admin_p.Web/Modules/electroDB/PromotionProducts/PromotionProductsForm.cs
using Serenity.ComponentModel;
using System;

namespace Admin_p.electroDB.Forms;

[FormScript("electroDB.PromotionProducts")]
[BasedOnRow(typeof(PromotionProductsRow), CheckNames = true)]
public class PromotionProductsForm
{
    public int PromotionId { get; set; }
    public int ProductId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/PromotionProductsForm.ts
import { ServiceLookupEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PromotionProductsForm {
    PromotionId: ServiceLookupEditor;
    ProductId: ServiceLookupEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
}

export class PromotionProductsForm extends PrefixedContext {
    static readonly formKey = 'electroDB.PromotionProducts';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PromotionProductsForm.init)  {
            PromotionProductsForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = DateEditor;

            initFormType(PromotionProductsForm, [
                'PromotionId', w0,
                'ProductId', w0,
                'CreatedAt', w1,
                'UpdatedAt', w1
            ]);
        }
    }
}